src/modules/PowerOCR/PowerOCRModuleInterface/dllmain.cpp

```diff
@@ -0,0 +1,199 @@
+    std::wstring app_key;
+
+    HANDLE send_telemetry_event;
+    HANDLE m_hInvokeEvent;
+    PROCESS_INFORMATION p_info;
+
+    bool is_process_running()
+    {
+        return WaitForSingleObject(p_info.hProcess, 0) == WAIT_TIMEOUT;
+    }
+
+    void launch_process()
+    {
+        Logger::trace(L"Launching PowerToys PowerOCR process");
+        unsigned long powertoys_pid = GetCurrentProcessId();
+
+        std::wstring executable_args = L"--use-pt-config --pid " + std::to_wstring(powertoys_pid);
+        std::wstring application_path = L"modules\\PowerOCR\\PowerToys.PowerOCR.exe";
+        std::wstring full_command_path = application_path + L" " + executable_args.data();
+        Logger::trace(L"PowerToys PowerOCR launching with parameters: " + executable_args);
+
+        STARTUPINFO info = { sizeof(info) };
+
+        if (!CreateProcess(application_path.c_str(), full_command_path.data(), NULL, NULL, true, NULL, NULL, NULL, &info, &p_info))
+        {
+            DWORD error = GetLastError();
+            std::wstring message = L"PowerToys PowerOCR failed to start with error: ";
+            message += std::to_wstring(error);
+            Logger::error(message);
+        }
+    }
+
+public:
+    // Constructor
+    PowerOCR()
+    {
+        // init_settings();
+    };
+
+    virtual void destroy() override
+    {
+        delete this;
+    }
+
+    virtual const wchar_t* get_name() override
+    {
+        return MODULE_NAME;
+    }
+
+    virtual bool get_config(wchar_t* buffer, int* buffer_size) override
+    {
+        HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+        PowerToysSettings::Settings settings(hinstance, get_name());
+        settings.set_description(MODULE_DESC);
+
+        return settings.serialize_to_buffer(buffer, buffer_size);
+    }
+
+    virtual const wchar_t* get_key() override
+    {
+        return app_key.c_str();
+    }
+
+    virtual void set_config(const wchar_t* config) override
+    {
+        try
+        {
+            // Parse the input JSON string.
+            PowerToysSettings::PowerToyValues values =
+                PowerToysSettings::PowerToyValues::from_json_string(config, get_key());
+
+            // If you don't need to do any custom processing of the settings, proceed
+            // to persists the values.
+            values.save_to_settings_file();
+        }
+        catch (std::exception&)
+        {
+            // Improper JSON.
+        }
+    }
+
+    virtual void enable()
+    {
+        ResetEvent(send_telemetry_event);
+        ResetEvent(m_hInvokeEvent);
+        launch_process();
+        m_enabled = true;
+    };
+
+    virtual void disable()
+    {
+        if (m_enabled)
+        {
+            Logger::trace(L"Disabling PowerOCR...");
+            ResetEvent(send_telemetry_event);
+            ResetEvent(m_hInvokeEvent);
+
+            // auto exitEvent = CreateEvent(nullptr, false, false, CommonSharedConstants::AWAKE_EXIT_EVENT);
```

Comment:
PowerOCR continues running after exiting Powertoys so probably this exit event is needed